api/v1/login.go

```diff
@@ -0,0 +1,61 @@
+package v1
+
+// LoginRequest is the request for the login request.
+type LoginRequest struct {
+	Token       string             `json:"token"`
+	Network     MachineNetworkSpec `json:"network"`
+	Location    MachineLocation    `json:"location"`
+	CPUInfo     MachineCPUInfo     `json:"cpuInfo"`
+	GPUInfo     MachineGPUInfo     `json:"gpuInfo"`
+	Provider    string             `json:"provider"`
+	MachineInfo MachineInfo        `json:"machineInfo"`
+}
+
+type MachineNetworkSpec struct {
+	PublicIP  string `json:"publicIP,omitempty"`
+	PrivateIP string `json:"privateIP,omitempty"`
+}
+
+type MachineLocation struct {
+	Region string `json:"region,omitempty"`
+	Zone   string `json:"zone,omitempty"`
+}
+
+type MachineCPUInfo struct {
+	Type         string `json:"type,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Architecture string `json:"architecture,omitempty"`
+}
+
+type MachineGPUInfo struct {
+	Product      string `json:"product,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Memory       string `json:"memory,omitempty"`
+}
+
+type MachineInfo struct {
+	BootID                  string `json:"bootID,omitempty"`
+	ContainerRuntimeVersion string `json:"containerRuntimeVersion,omitempty"`
+	KernelVersion           string `json:"kernelVersion,omitempty"`
+	OperatingSystem         string `json:"operatingSystem,omitempty"`
+	OSImage                 string `json:"osImage,omitempty"`
+	Uptime                  string `json:"uptime,omitempty"`
+	GPUDriverVersion        string `json:"gpuDriverVersion,omitempty"`
+	CUDAVersion             string `json:"cudaVersion,omitempty"`
+	// MachineID is the ID of the machine collected from GPUd
+	MachineID string `json:"machineID,omitempty"`
+}
+
+// LoginResponse is the response for the login request.
+type LoginResponse struct {
+	// Status is the status of the login request.
+	Status string `json:"status"`
+
+	// MachineID is the ID of the machine generated by the login control plane.
+	// Only set upon successful login.
+	MachineID string `json:"machine_id,omitempty"`
```

Comment:
Let's keep the same style.. 
```suggestion
	MachineID string `json:"machineID,omitempty"`
```

Reply:
@cardyok ^